File: docs/examples/basic-inbound.ts
/**
 * Basic inbound call handler.
 *
 * The AI answers incoming calls and responds to what the caller says.
 * Replace the API key with your own from https://api.patter.dev.
 *
 * Usage:
 *   npm install patter
 *   npx ts-node basic-inbound.ts
 */

import { Patter, IncomingMessage } from "patter";

const phone = new Patter({ apiKey: "pt_your_api_key_here" });

async function onMessage(msg: IncomingMessage): Promise<string> {
  console.log(`Caller said: "${msg.text}"`);

  const text = msg.text.toLowerCase();

  if (text.includes("hours") || text.includes("open")) {
    return "We are open Monday through Friday, 9 AM to 6 PM Eastern time.";
  }

  if (text.includes("help") || text.includes("support")) {
    return "I can help you with billing, technical questions, or account changes. What do you need?";
  }

  if (text.includes("bye") || text.includes("goodbye")) {
    return "Thanks for calling. Have a great day! Goodbye.";
  }

  return `You said: ${msg.text}. How can I help you today?`;
}

async function main(): Promise<void> {
  console.log("Connecting to Patter...");

  await phone.connect({
    onMessage,
    onCallStart: async (data) => {
      console.log(`Incoming call from ${data.caller ?? "unknown"}`);
    },
    onCallEnd: async () => {
      console.log("Call ended");
    },
  });

  console.log("Ready. Waiting for incoming calls... (Ctrl+C to stop)");

  // Keep the process alive
  await new Promise<never>(() => {});
}

process.on("SIGINT", async () => {
  console.log("\nShutting down...");
  await phone.disconnect();
  process.exit(0);
});

main().catch(console.error);

File: docs/examples/basic-inbound.py
"""
Basic inbound call handler.

The AI answers incoming calls and responds to what the caller says.
Replace the API key with your own from https://www.getpatter.com.

Usage:
    pip install getpatter
    python basic-inbound.py
"""

import asyncio
from patter import Patter, IncomingMessage


async def on_message(msg: IncomingMessage) -> str:
    """Called for every utterance the caller makes. Return what the AI should say."""
    print(f"Caller said: {msg.text!r}")

    text = msg.text.lower()

    if "hours" in text or "open" in text:
        return "We are open Monday through Friday, 9 AM to 6 PM Eastern time."

    if "help" in text or "support" in text:
        return "I can help you with billing, technical questions, or account changes. What do you need?"

    if "bye" in text or "goodbye" in text:
        return "Thanks for calling. Have a great day! Goodbye."

    return f"You said: {msg.text}. How can I help you today?"


async def on_call_start(data: dict) -> None:
    print(f"Incoming call from {data.get('caller', 'unknown')}")


async def on_call_end(data: dict) -> None:
    print("Call ended")


async def main() -> None:
    phone = Patter(api_key="pt_your_api_key_here")

    print("Connecting to Patter...")
    await phone.connect(
        on_message=on_message,
        on_call_start=on_call_start,
        on_call_end=on_call_end,
    )
    print("Ready. Waiting for incoming calls... (Ctrl+C to stop)")

    try:
        await asyncio.Event().wait()
    except KeyboardInterrupt:
        print("\nShutting down...")
        await phone.disconnect()


if __name__ == "__main__":
    asyncio.run(main())
